File: DTC.Api/Dtos/Groups/GroupDto.cs
using DTC.Api.Dtos.GroupPlayer;
using DTC.Api.Dtos.Round;
using DTC.Api.Models; 

namespace DTC.Api.Dtos.Groups
{
    public class GroupDto
    {
        public int Id { get; set; }
        public int TournamentId { get; set; }
        public int BoardId { get; set; }
        public int Sequence { get; set; }
        public string Name { get; set; } = string.Empty;
        public int QualifiersCount { get; set; }

        public ICollection<GroupPlayerDto> Players { get; set; } = new List<GroupPlayerDto>();
        public ICollection<RoundDto> Rounds { get; set; } = new List<RoundDto>();
    }
}

File: dtc.frontend/src/dtos/group/GroupDto.ts
import z from "zod";

export const groupDtoSchema = z.object({
  id: z.number(),
  tournamentId: z.number(),
  sequence: z.number(),
  name: z.string(),
  qualifiersCount: z.number(),
});

export type GroupDto = z.infer<typeof groupDtoSchema>;
